crates/utils/src/utils/markdown.rs

```diff
@@ -1,3 +1,131 @@
+use markdown_it::MarkdownIt;
+
+pub mod spoiler_rule;
+
 pub fn markdown_to_html(text: &str) -> String {
-  comrak::markdown_to_html(text, &comrak::ComrakOptions::default())
+  let md = &mut MarkdownIt::new();
+  markdown_it::plugins::cmark::add(md);
+  markdown_it::plugins::extra::add(md);
+  spoiler_rule::add(md);
```

Comment:
It's unfortunate that the parser needs to be constructed with each request; I attempted to place this into a static context but was unsuccessful due to how this builder is set up. Open to suggestions on how to make this into a singleton.

Reply:
Use [once_cell](https://docs.rs/once_cell/latest/once_cell/index.html) to store things in static context. Some of this was also moved to standard library recently.

Reply:
I gave this a go, but kept running into issues around ownership since `MarkdownIt` has to be mutable based on the implementation. The tests run relatively fast and it looks like `cromak` is similarly sized under the hood, so maybe this is OK as is?

Reply:
Have you tried wrapping it in Arc?

Reply:
The very nice folks on the Rust discord helped me out here 🙌 Just an issue with my understanding of mutable references.